### PR TITLE
Add 2014 - 2020 and 2023-2024 Arkansas income tax rate

### LINKED ISSUE #3826: Add 2014 - 2020 and 2023-2024 Arkansas income tax rate
[Law](https://casetext.com/statute/arkansas-code-of-1987/title-26-taxation/subtitle-5-state-taxes/chapter-51-income-taxes/subchapter-2-imposition-of-tax/section-26-51-201-individuals-trusts-and-estates) - 2023 and on

[2014 -2020 law](https://codes.findlaw.com/ar/title-26-taxation/ar-code-sect-26-51-201/)

### PR DESCRIPTION
**Pull request by YichunRen**
Fixes #3826

### CODE REVIEW

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 1]
```diff
@@ -12,44 +12,64 @@ metadata:
...
+
 brackets:
   - threshold:
       2021-01-01: 0
+      2023-01-01: 0
+      2024-01-01: 0
     amount:
       2021-01-01: 0
+      2023-01-01: 0
+      2024-01-01: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:28]:
```suggestion
```

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 2]
```diff
@@ -12,44 +12,64 @@ metadata:
...
+  - title: 2023-2024 Indexed Tax Brackets
+    href: https://casetext.com/statute/arkansas-code-of-1987/title-26-taxation/subtitle-5-state-taxes/chapter-51-income-taxes/subchapter-2-imposition-of-tax/section-26-51-201-individuals-trusts-and-estates
+  - title: 2014-2020 Indexed Tax Brackets
+    href: https://codes.findlaw.com/ar/title-26-taxation/ar-code-sect-26-51-201/
+
 brackets:
   - threshold:
       2021-01-01: 0
+      2023-01-01: 0
+      2024-01-01: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:24]:
```suggestion
```

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 3]
```diff
@@ -12,44 +12,64 @@ metadata:
...
     amount:
       2021-01-01: 0
+      2023-01-01: 0
+      2024-01-01: 0
   - threshold:
       2021-01-01: 4_800
       2022-01-01: 5_100
-      2023-01-01: 5_300
+      2023-01-01: 5_100
+      2024-01-01: 5_100
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:33]:
```suggestion
```

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 4]
```diff
@@ -12,44 +12,64 @@ metadata:
...
+      2024-01-01: 0
     amount:
       2021-01-01: 0
+      2023-01-01: 0
+      2024-01-01: 0
   - threshold:
       2021-01-01: 4_800
       2022-01-01: 5_100
-      2023-01-01: 5_300
+      2023-01-01: 5_100
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:32]:
```suggestion
```

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 5]
```diff
@@ -12,44 +12,64 @@ metadata:
...
+      2024-01-01: 5_100
     amount:
       2021-01-01: 0.02
+      2023-01-01: 0.02
+      2024-01-01: 0.02
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:37]: dont need values if they do not change, just the ones that do. 
if for example
2022-01-01: 10,000
then every year from then o (2023, 2024, ... , 2050) are all 10,000 unless we specify it other wise, so we only need to add values if they change

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Add 2014 - 2020 and 2023-2024 Arkansas income tax rate
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
- bump: patch
  changes:
    fixed:
    - Arkansas income tax rates.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 7]
```diff
@@ -5,51 +5,93 @@ metadata:
...
+
 brackets:
   - threshold:
-      2021-01-01: 0
+      2014-01-01: 0
     amount:
-      2021-01-01: 0
+      2014-01-01: 0.009 
+      2019-01-01: 0
   - threshold:
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:29]: please add uprating while here

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 8]
```diff
@@ -5,51 +5,99 @@ metadata:
...
+        rounding:
+          type: downwards
+          interval: 100
   - threshold:
+      2014-01-01: 4_300
+      2019-01-01: 4_600
+      2020-01-01: 4_700
       2021-01-01: 4_800
       2022-01-01: 5_100
       2023-01-01: 5_300
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:41]:
```suggestion
  - threshold:
      values;
        2014-01-01: 4_300
        2019-01-01: 4_600
        2020-01-01: 4_700
        2021-01-01: 4_800
        2022-01-01: 5_100
        2023-01-01: 5_300
    metadata:
      uprating: gov.irs.uprating
      rounding:
        type: downwards
        interval: 100
```

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 9]
```diff
@@ -5,51 +5,99 @@ metadata:
...
     amount:
-      2021-01-01: 0
+      2014-01-01: 0.009 
+      2019-01-01: 0
+      # one rate for multiple threshold values??
+      uprating:
+        parameter: gov.irs.uprating
+        rounding:
+          type: downwards
+          interval: 100
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:34]: The uprating should only be applied to the thresholds not the rates, see example below

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 10]
```diff
@@ -5,51 +5,99 @@ metadata:
...
+
 brackets:
   - threshold:
-      2021-01-01: 0
+      2014-01-01: 0
     amount:
-      2021-01-01: 0
+      2014-01-01: 0.009 
+      2019-01-01: 0
+      # one rate for multiple threshold values??
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:29]:
```suggestion
```

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml (hunk 8 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:41]: this format for each threshold

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/reduction.yaml [hunk 11]
```diff
@@ -425,14 +425,4 @@ brackets:
...
-      2021-01-01: .inf
-    amount:
-      2021-01-01: 247.5
-      2022-01-01: 167.20
-      2023-01-01: .inf
-  - threshold:
-      2021-01-01: .inf
-    amount:
-      2021-01-01: .inf
+      2023-01-01: 153.70
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/reduction.yaml:428]: do we need a .inf bracket for 2022?

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 12]
```diff
@@ -1,55 +1,140 @@
 description: Arkansas taxes personal income at these amounts.
 metadata:
-  type: single_amount
+  type: single_amount 
   threshold_unit: currency-USD
   amount_unit: /1
   label: Arkansas income tax amount
+  period: year
   reference:
-  - title: Tax Legislation for 2022 Ark. Code § 26-51-201 (a)(1)(A)
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:8]: why remove the legal code reference? is there another we can include?

Comment (YichunRen) [reply policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:8]: The legal code reference has been re-added to the reference list.

File: policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml [hunk 13]
```diff
@@ -1,55 +1,142 @@
 description: Arkansas taxes personal income at these amounts.
 metadata:
-  type: single_amount
+  type: single_amount 
   threshold_unit: currency-USD
   amount_unit: /1
   label: Arkansas income tax amount
+  period: year
   reference:
   - title: Tax Legislation for 2022 Ark. Code § 26-51-201 (a)(1)(A)
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ar/tax/income/rates/main/rate.yaml:9]:
```suggestion
  - title: Tax Legislation for 2022 Ark. Code § 26-51-201 (a)(1)(A) # (d)(1) for uprating.
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**